File: soft-update-and-hard-delete/src/main/java/com/app/entity/Member.java
package com.app.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Entity
@DynamicInsert
@Where(clause = "status = 'Y'")
@SQLDelete(sql = "UPDATE member SET status = 'N' WHERE member_id = ?")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Long id;

    @Column(unique = true, length = 50, nullable = false)
    private String email;

    @ColumnDefault("'Y'")
    @Column(length = 1)
    private String status;

    @OneToMany(mappedBy = "member")
    private List<Content> content = new ArrayList<>();

    public Member(String email) {
        this.email = email;
    }
}

File: kotlin_spring_multi_module/storage/database-core/src/main/kotlin/app/learning/storage/db/core/member/Member.kt
package app.learning.storage.db.core.member

import app.learning.storage.db.core.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Member(
    val email: String,
    password: String,
    name: String,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
) : BaseEntity() {

    var password: String = password
        protected set

    var name: String = name
        protected set

}